src/main/java/com/thealgorithms/conversions/BinaryToOctal.java

```diff
@@ -31,20 +20,26 @@ public static void main(String[] args) {
      * @return The octal number
      */
     public static String convertBinaryToOctal(int binary) {
-        String octal = "";
-        int currBit = 0;
-        int j = 1;
+        if (binary == 0) {
+            return "0";
+        }
+
```

Comment:
Consider adding input validation in the convertBinaryToOctal method to ensure that the input is a valid binary number (composed only of 0 and 1). If the input is invalid, the method could throw an IllegalArgumentException or handle it gracefully by returning an appropriate error message.

`if (!String.valueOf(binary).matches("[01]+")) {
    throw new IllegalArgumentException("Input is not a valid binary number.");
}
`

Reply:
Good observation. Thanks!